### DISCUSSION TITLE
IDASolver failed

### DISCUSSION THREAD

**Discussion by zmdntnu** [do-mpc/do-mpc#226] [Q&A] [answered]
Dear @4flixt,

Hi, I came into a new issue like this:

CasADi -  WARNING("simulator:jacF failed: NaN detected for output o0, at nonzero index 8 (row 8, col 3).") [.../casadi/core/oracle_function.cpp:265]
psetup failed: .../casadi/interfaces/sundials/idas_interface.cpp:849: Calculating Jacobian failed
CasADi -  WARNING("simulator:daeF failed: NaN detected for output alg, at (row 0, col 0).") [.../casadi/core/oracle_function.cpp:265]
CasADi -  WARNING("simulator:jacF failed: NaN detected for output o0, at nonzero index 8 (row 8, col 3).") [.../casadi/core/oracle_function.cpp:265]
psetup failed: .../casadi/interfaces/sundials/idas_interface.cpp:849: Calculating Jacobian failed
At t = 0.323855, , mxstep steps taken before reaching tout.
Traceback (most recent call last):
  File "D:/python file'/MPC_planning/example_4.py", line 330, in <module>
    y_next = simulator.make_step(u0)
  File "D:\python file'\venv\lib\site-packages\do_mpc\simulator.py", line 500, in make_step
    x_next = self.simulate()
  File "D:\python file'\venv\lib\site-packages\do_mpc\simulator.py", line 431, in simulate
    r = self.simulator(x0 = sim_x_num, z0 = sim_z_num, p = sim_p_num)
  File "D:\python file'\venv\lib\site-packages\casadi\casadi.py", line 13453, in __call__
    return self.call(kwargs)
  File "D:\python file'\venv\lib\site-packages\casadi\casadi.py", line 12324, in call
    return _casadi.Function_call(self, *args)
RuntimeError: .../casadi/interfaces/sundials/idas_interface.cpp:591: IDASolve returned "IDA_TOO_MUCH_WORK". Consult IDAS documentation.

Process finished with exit code 1

But I couldn't locate the cause of such issue. Could you please take a look at it when you are available? Thank you so much.

Best wishes

Zhu mingda

**Comment by 4flixt** [comment, marked as answer]
The important part here is ``NaN detected``. You are most likely trying to simulate an ODE / solve an optimization problem, where you give initial values / initial guesses that lead to e.g. the division by zero (yielding Not a Number = NaN). Gradients cannot be computed for this point and thus the simulator / solver fails.
Please check you equations (state dynamics etc.) and identify if there is an operation (e.g. division, log, ...) that would yield NaN for some values.
